unstructured/partition/text.py

```diff
@@ -0,0 +1,64 @@
+import email
+import re
+from typing import Dict, IO, List, Optional
+
+from unstructured.cleaners.core import replace_mime_encodings
+from unstructured.documents.elements import Element, Text
+from unstructured.partition.text_type import (is_possible_narrative_text
+                                              ,is_possible_title
+                                              ,is_bulleted_text
+                                             )
+
+
+def partition_text(
+    filename: Optional[str] = None, file: Optional[IO] = None, text: Optional[str] = None
+) -> List[Element]:
+    """Partitions an .eml documents into its constituent elements.
+    Parameters
+    ----------
+     filename
+        A string defining the target filename path.
+    file
+        A file-like object using "r" mode --> open(filename, "r").
+    text
+        The string representation of the .eml document.
+    """
+    if not any([filename, file, text]):
+        raise ValueError("One of filename, file, or text must be specified.")
+
+    if filename is not None and not file and not text:
+        with open(filename, "r") as f:
+            msg = email.message_from_file(f)
+
+    elif file is not None and not filename and not text:
+        file_text = file.read()
+        msg = email.message_from_string(file_text)
+
+    elif text is not None and not filename and not file:
+        _text: str = str(text)
+        msg = email.message_from_string(_text)
+
+    else:
+        raise ValueError("Only one of filename, file, or text can be specified.")
+
+    content_map: Dict[str, str] = {
+        part.get_content_type(): part.get_payload() for part in msg.walk()
+    }
+    content = content_map.get("text/plain", "")
+    if not content:
+        raise ValueError("text/plain content not found in email")
+
+    content = re.split(r"\n\n\n|\n\n|\n", content)
```

Comment:
I'd make `split_by_paragraph` a separate helper function and also `\r` and other line-ending variants a well. The links below provide some background on that.

- https://stackoverflow.com/questions/20056306/match-linebreaks-n-or-r-n
- https://stackoverflow.com/questions/1761051/difference-between-n-and-r